src/authgss_hash.c

```diff
@@ -126,16 +129,10 @@ authgss_hash_init()
 	    __svc_params->gss.max_ctx / authgss_hash_st.xt.npart;
 	authgss_hash_st.initialized = true;
 
+unlock:
 	mutex_unlock(&authgss_hash_st.lock);
 }
 
-#define cond_init_authgss_hash() { \
-		do { \
-			if (!authgss_hash_st.initialized) \
-				authgss_hash_init(); \
-		} while (0); \
-	}
-
```

Comment:
This is done without the lock for speed. It is better to keep this macro as the boolean here can only go from false to true (never from true to false). Please keep this optimization.